src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/Program.cs

```diff
@@ -94,80 +90,44 @@ private static void Execute(string[] args, bool isDispatcher, bool isNoBuild, IL
-                        configuration,
-                        isNoBuild,
-                        logger);
-                }
-                else
-                {
-                    if (!TryGetNugetFramework(framework.Value(), out nugetFramework))
-                    {
-                        throw new ArgumentException($"Could not understand the NuGetFramework information. Framework short folder name passed in was: {framework.Value()}");
-                    }
-
-                    var nearestNugetFramework = NuGetFrameworkUtility.GetNearest(
-                        frameworksInProject,
-                        nugetFramework,
-                        f => new NuGetFramework(f));
-
-                    if(nearestNugetFramework == null)
-                    {
-                        // This should never happen as long as we dispatch correctly.
-                        var msg = "Could not find a compatible framework to execute."
-                            + Environment.NewLine
-                            +$"Available frameworks in project:{string.Join($"{Environment.NewLine} -", frameworksInProject.Select(f => f.GetShortFolderName()))}";
-                        throw new InvalidOperationException(msg);
-                    }
-
-                    ProjectContext context = new ProjectContextBuilder()
-                        .WithProject(projectFile)
-                        .WithTargetFramework(nearestNugetFramework)
-                        .Build();
-
-                    Debug.Assert(context != null);
-
-                    var codeGenArgs = ToolCommandLineHelper.FilterExecutorArguments(args);
-
-                    CodeGenCommandExecutor executor = new CodeGenCommandExecutor(
-                        context,
-                        codeGenArgs,
-                        configuration,
-                        packagesPath.Value(),
-                        logger);
-
-                    return executor.Execute();
-                }
+                DieIfProjectJson(project);
+
+                // Invoke the tool from the project's build directory.
+                return BuildAndDispatchDependencyCommand(
+                    args,
+                    project,
+                    buildBasePath.Value(),
+                    configuration,
+                    isNoBuild,
+                    logger);
+
             });
 
             app.Execute(args);
         }
 
+        private static void DieIfProjectJson(string projectPath)
+        {
+            var attr = File.GetAttributes(projectPath);
+            if (projectPath.EndsWith("project.json")
+                && !attr.HasFlag(FileAttributes.Directory))
+            {
+                var msg = $"This version of {TOOL_NAME} does not support project.json based projects.{Environment.NewLine}"
+                    +$"Please migrate your project to newer version, or downgrade the version of {TOOL_NAME} to {PROJECT_JSON_SUPPORT_VERSION}";
```

Comment:
I don't really see the variables being in multiple places. Can they just be part of this string directly?